tchannel-core/src/main/java/com/uber/tchannel/api/SubChannel.java

```diff
@@ -116,7 +122,7 @@ public String getServiceName() {
         return requestHandlers.get(endpoint);
     }
 
-    public Connection.Direction getPreferredDirection() {
+    public @NotNull Connection.Direction getPreferredDirection() {
```

Comment:
there's no evidence this must be non-null (though it might've been the original intent).